### ISSUE TITLE
Using return value of a function that returns None is not always reported.

### ISSUE THREAD

**Issue by NielsPT** [microsoft/pyright#10767]
## Environment data

-   Pylance version: 2025.7.1
-   OS and version: MacOS 15.5 (24F74)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.13.5

## Code Snippet

```python
def f1(i: int) -> None:
    print(i)


def f2() -> None:
    r = f1(2)  # assignment-from-no-return
    if not r:
        print("error")
    if not f1(2):  # no error
        print("error")
```

## Repro Steps

Insert code above in editor. Note that there is no squiggly on line 9.

## Expected behavior

Error reported on line 9.

## Actual behavior

No error reported.

## Logs

NA

## Description

I would expect the behaviour of line 6 and line 9 of the example above to be the same. This might be by design, but that would, in my opinion, be a bad design. As I see it, taking the return value of a function that returns None should always be flagged as an error, no matter whether the value is assigned to a variable or used directly in an if-statement or similar.
Whether this issue is a feature request or a bug, I'm not sure. Feel free to change it accordingly.

**Comment by erictraut** [comment]
Pyright is working as intended here. 

A return type of `None` is legitimate, and there is no type violation here, so a type checker like pyright will not (and should not, IMO) flag this as an error. What you're asking for here is more of a linting check. I recommend checking out ruff, pylint, or other linters to see if they can enforce the rule that you're asking for here. 

The `assignment-from-no-return` rule that you mentioned in your code sample appears to be a [pylint rule](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/assignment-from-no-return.html), so I'm guessing that you're already using pylint in your project. If you'd like to see that rule expanded, please file a feature request in the pylint project.
